src/include/daos_uns.h

```diff
@@ -161,6 +161,34 @@ struct duns_attr_t {
 int
 duns_create_path(daos_handle_t poh, const char *path, struct duns_attr_t *attrp);
 
+/**
+ * Create a special directory (POSIX) or file (HDF5) depending on the container type, and create a
+ * new DAOS container with a givren set of user attributes. The uuid of the container can be either
+ * passed in \a attrp->da_cuuid (deprecated) or generated internally and returned in \a da_cont.
+ * The extended attributes are set on the dir/file created that points to pool uuid, container
+ * uuid. This is to be used in a unified namespace solution to be able to map a path in the unified
+ * namespace to a location in the DAOS tier.  The container and pool can have labels, but the UNS
+ * stores the uuids only and so labels are ignored in \a attrp.
+ * The user is not required to call duns_destory_attrs on \a attrp as this call does not allocate
+ * any buffers in attrp.
+ *
+ * \param[in]	poh	Pool handle
+ * \param[in]	path	Valid path in an existing namespace.
+ * \param[in]	count	Number of container attributes
+ * \param[in]	names	Array of \a count null-terminated container attribute names.
+ * \param[in]	values	Array of \a count container attribute values
+ * \param[in]	sizes	Array of \a count elements containing the sizes of respective container
+ *			attribute values.
+ * \param[in,out]
+ *		attrp	Struct containing the attributes. The uuid of the
+ *			container created is returned in da_cuuid.
+ *
+ * \return		0 on Success. errno code on failure.
+ */
+int
+duns_create_path_attr(daos_handle_t poh, const char *path, int count, char const *const names[],
+		      void const *const values[], size_t const sizes[], struct duns_attr_t *attrp);
```

Comment:
im not a fan of this new API. it might be useful to have something for the command line.
but for the API, it's not a big deal to do
daos_cont_create()
daos_cont_set_attr()
I would suggest to remove it.

Reply:
Not sure to fully understand your comments.
Are you suggesting to remove `duns_create_path_attr()` and `duns_create_path()` functions or just the first one.
If we are just removing `duns_create_path_attr()`, then we will have two solutions from my understanding:
1) Change the dfuse code to poll the attributes of its mounted containers.
2) split the function `duns_create_path_attr()` to not create and bind the container in the same function.  By this way, we will be able to use the function `daos_cont_set_attr()` between the two function calls.

Reply:
no you cannot remove duns_create_path()!
my question is why you are adding this new API (duns_crreate_path_attr()) ?
what is the challenge of doing:
duns_create_path()
daos_cont_set_attr()
?

Reply:
From my understanding, doing this will not work as the attributes are cached inside DFuse after the binding.
Thus, when we will get back from `duns_create_path()` , the call to `daos_cont_set_attr()` will not be taken into account by DFuse.
@ashleypittman could you confirm this.

Reply:
I've taken a closer look and think you're both right, having a new API doesn't seem like the right solution here, however I hadn't realised it was the same C function that both created the container and inserted it into the POSIX namespace.

The best path might be:
daos_cont_create()
daos_cont_set_attr()
duns_link_cont()

Reply:
I was not aware of the duns_link_cont() function, which for sure is a better solution.
Thanks @ashleypittman for the hints and @mchaarawi for the relevant remark :-)

- [ ] Remove useless new function `duns_create_path_attr()`
- [ ] Split sub container creation

Reply:
> I was not aware of the duns_link_cont() function, which for sure is a better solution. Thanks @ashleypittman for the hints and @mchaarawi for the relevant remark :-)
> 
> * [x]  Remove useless new function `duns_create_path_attr()`
> * [x]  Split sub container creation

Fixed with commit 819b856455ea7ad9a0fe3e3bb0d851aaa5b10afc